airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -361,11 +367,27 @@ public WebBackendConnectionRead webBackendUpdateConnection(final WebBackendConne
     connectionRead = connectionsHandler.updateConnection(connectionUpdate);
 
     if (needReset) {
+      final UUID connectionId = webBackendConnectionUpdate.getConnectionId();
+      final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody().connectionId(connectionId);
+      final ConnectionStateType stateType = getStateType(connectionIdRequestBody);
+      final List<io.airbyte.protocol.models.StreamDescriptor> streamsToReset;
+
+      if (stateType == ConnectionStateType.LEGACY || stateType == ConnectionStateType.NOT_SET) {
```

Comment:
As discussed on slack [here](https://airbytehq-team.slack.com/archives/C03AS1GAQV6/p1656537227955969?thread_ts=1656533917.939489&cid=C03AS1GAQV6), we should also reset all streams in the `GLOBAL` case for now, until we have confirmation that per-stream resets can actually be performed for CDC connectors